### ISSUE TITLE
~/assets alias doesn't work inside @import but relatives paths works

### ISSUE THREAD

**Issue by gtso86** [nuxt/nuxt.js#1407]
Hey guys, I found a bug and I don't know if it is related with Nuxt and/or Vue-loader.

Since Nuxt uses [Single File Components](https://vuejs.org/v2/guide/single-file-components.html) I usually create a global css with all my variables and import inside the `<style></style>` tag but seems the `~/assets` or `~assets` isn't working properly.

I got a error when I try to use it:
![error](http://i.imgur.com/tbwTBAy.png)

BUT when I used **relative path** works fine
```css
<style>
  /* works! */
  @import "../../assets/styles/element/theme-variables.css";
  /* fails  */
  @import "~assets/styles/element/theme-variables.css";

   :root {
    --sidebar-padding: var(--notification-padding);
  }

  .login {
    background-image: url(~assets/images/waves.png);
    background-position: 0% 110%;
    background-repeat: no-repeat;
  }

  .login-sidebar {
    background: var(--dark-menu-item-fill);
    padding: var(--sidebar-padding);
    height: 100%;
    width: 400px;
    color: var(--color-white);

    & .el-form-item__label {
      color: var(--color-white);
    }
  }
</style>
```

I'm following this thread #1241 and looks like with the next release (RC5) the ~/assets (eg: used for background-images) will fixed too. 

>_**Workaround** or the right way?_

On my previous projects using Vue-loader,  I was using the [sass resources loader](https://github.com/shakacode/sass-resources-loader) to accomplish something similar, serving bootstrap variables file across all *.scss and inside *.vue templates,  that way, sass-loader have all variables available before generate the compiled file.

Now, I'm using a different set up (postCSS + CSSnext) and after some time struggling to configure, I archived what I need it with few pitfalls.

I used this thread #1100 as base to set up.

---

**The question is:** the only way to serve global files is using this workaround OR relatives paths?

**Comment by gtso86** [comment]
#1100

**Comment by pi0** [comment]
Hey @gtso86 thanks for your good detailed issue reports (I hope everyone reports this way). Actually with rc5 the default postCSS setup will be CSSNext. Also if look at new [style-loader.js](https://github.com/nuxt/nuxt.js/blob/dev/lib/builder/webpack/style-loader.js) it adds [resolve-url-loader](https://github.com/bholloway/resolve-url-loader) to the chain and there is a [postcss-import](https://github.com/nuxt/nuxt.js/blob/dev/lib/common/options.js#L90) enabled in the default config. Actually, it may need more work for using with plain css `@import`s.

**Comment by pi0** [comment]
With  a721232 postcss-import would like everywhere like this:

```css
@import "assets/styles/element/theme-variables.css";
```

For supporting `~/` prefix we may need to fork or improve its `resolve` mechanism.

**Comment by gtso86** [comment]
> thanks for your good detailed issue reports

This is the least I can do! This is my first time diving in to learn something after pos-jQuery era and I'm loving how Vue/Nuxt is simple and powerful!

Since I can't help contributing with PR because I'm still learning how to deal with all this new things (ES6/WebPack/Modules - those things are killing me hahah) every bug I found I tried to reproduce in a new env and report in here with all details I can get.

---

_[off topic]_

I know the documentation still in progress/evolving since every week or so a new release is lunched. But one thing always helps me are [examples](https://nuxtjs.org/examples) and real case snippets. Even for small things like how to change the server port, etc..

I guess examples could reduce GitHub to be flooded with common/daily use questions.

**Comment by gtso86** [comment]
>Also if look at new style-loader.js it adds resolve-url-loader to the chain and there is a postcss-import enabled in the default postcss config.

I tried use **postcss-import** before moved to **sass resources loader** workaround because I got some side effects...

The main problem was because I was using sass resources loader + postcss-import + postCSS to render every .css/.scss/.vue file.

Since .css files only allow you to use @import at the first line from each file, my main.css stopped to works as expected because sass-resources OR postcss-import was injecting the global variables before my @import rules...

to work around the problem I changed my main.css to main.scss and removed the postcss from .scss files. and to avoid postcss process every @import rule I removed it postcss-import.

The other problem I faced was because postcss-import was rendering multiple times my variables.css inside every file... my sourcemaps became a mess! And because I have the same rules multiple times was painful use devtools to debug.
